### PR TITLE
Changed the handling of Table Defs and renamed functions to prevent conflicts.

### PR DESCRIPTION
**Pull request by seanchase**
The major changes are centered around (Im/Ex)portTableDef, removing unused code, and renaming all public functions outside "VCS_ImportExport" by appending "VCS_" to them.

TableDefs are now stored in XML Table Schema instead of SQL text files. This is HUGE because you no longer need to add ADO references and can import directly into existing AccessDBs.

With new TableDefs functions there were a couple functions no longer needed. I scrubbed the code and found several more and removed them as well.

As far as renaming the other functions goes, I was getting several naming conflicts with existing code. "FileExists" function being the obvious example. Adding VCS_ to everything looks ugly but makes the project more robust. It shrinks the conflict target down to 4 functions. Main() is still very vulnerable to this.

In my opinion the project is all 'Guest' code serving an existing project. Eventually, I'd like to see everything go back to one module so as not to crowd out the already busy modules list. It should be able to be dropped into any Access Database and compile without conflict. The real work people should be doing is getting their project on GitHub or other VCS and start collaborating. Not getting the damn thing to work.

### CODE REVIEW

**Comment by ubidev** [comment]
Congrats Sean for your work! 

On what versions of Access were your mods tested? Won't the new xml export code replacement break support for some old Access versions already supported by this project in its current state?

btw, I'm totally with you on your wishes for the future directions of this project
cheers :)

ubidev

**Comment by timabell** [comment]
Cheers for the PR, I'll leave it open for a while for the community to comment on. I'm not spending much time reviewing contributions so basically if there's no objections raised it'll go in.

**Comment by seanchase** [comment]
I'm using access 2010. It should work as far back as 2007 I believe. Anything older than that is likely not supported anymore. If you can open .accdb files, this should work.

**Comment by mgrojo** [comment]
Great work, but I see a drawback for my use of the table xml export feature. I don't use the import feature. I've only use it for testing this project. What I like of the SQL export is that it's easy to read and understand. I only use this software to control the changes I make to my project, so I wouldn't like very much to lose the SQL export, because XML is less readable and it will be disruptive to the version history of the controlled project. Could we preserve the SQL export, although the import is made from the XML, if it works better?

**Comment by seanchase** [comment]
While it is possible to leave the old code and have SQL and schema output , the biggest issue is that brings back the terrible ADO library dependencies. My original idea to be less disruptive was to output the xml, parse out the DECIMAL values needed, and use that instead of the function with the dependencies. The moment I started to code I figured why bother? It shows up as 4 lines for clarity, but in reality its 1 line of code that replaces at least 100+ and is much more stable. And the import is saves another 70+. 
How much code it replaced is not important. The significance is it's stable and uses built in, documented, functionality. I see the value in having it in SQL as well. It would give me a reason to learn how to parse XML 😁. 

TLDR: new way is better, it will take some work to add it back.

**Comment by FunkyM** [comment]
Great stuff, a few notes on the patches:
* ImportAllSource() fails loading of table data due to missing schemas. The code before only consideres "*.sql" files when loading the table schemas, but the "*.xml" schemas should be loaded instead. Probably a typo.
* The sanitizer corrupts form sources due to a logic error in the way the indent regex code works.
* The VCS_UsingUcs2() detection does not work properly, yielding True despite UTF-8 for instance, which leads to corrupt files that Access fails to import using LoadFromText().
* The Sleep() Sub is missing the PtrSafe attribute for 64bit Office.
* I'd also prefix the public Sub/Functions with VCS_ so it's clear these are called from the VCS modules.

After fixing the stuff things are working pretty neatly and it's nice to get rid of the library dependencies.
The code overall could need some more normalization. The sanitizer performance could also need some optimization. Might try to work on that if time allows...

I am using git and a dev database that gets versioned instead of the "final project database" that acts as a blank template and upon opening it, it uses the VCS code to completely initialize itself from the sources. Deployment is simply done by saving the current state to a new file.

That way everything is truly versioned, multiple developers can work on one project and no big binary project database blob needs to be committed.

**Comment by FunkyM** [comment]
Created a PR for mentioned fixes seanchase/msaccess-vcs-integration#1. Works fine for me now.

**Comment by chvndb** [comment]
Any reason why this hasn't been merged yet? I have been using @FunkyM branch and seems to be working so far.

**Comment by timabell** [comment]
When I said a while, I apparently meant a couple of years... Do feel free to kick me if anything languishes, it's sometimes easy to forget if you don't act on the github notification immediately.
